torchvision/datasets/imagenet.py

```diff
@@ -118,54 +107,115 @@ def extra_repr(self):
         return "Split: {split}".format(**self.__dict__)
 
 
-def parse_devkit(root):
-    idx_to_wnid, wnid_to_classes = parse_meta(root)
-    val_idcs = parse_val_groundtruth(root)
-    val_wnids = [idx_to_wnid[idx] for idx in val_idcs]
-    return wnid_to_classes, val_wnids
+@contextmanager
```

Comment:
Can we move this to utils.py?

Reply:
I think this is fine. We will be moving to Python 3 together with PyTorch early 2020, so we will be able to use `tempfile.TemporaryDirectory` instead